src/coreclr/src/tools/crossgen2/crossgen2/crossgen2.csproj

```diff
@@ -40,30 +40,68 @@
     </PackageReference>
   </ItemGroup>
 
-  <PropertyGroup Condition="'$(TargetsUnix)' == 'true'">
+  <PropertyGroup>
+    <CrossHostArch></CrossHostArch>
+    <CrossHostArch Condition="'$(TargetArchitecture)' == 'arm64'">x64</CrossHostArch>
+
+    <TargetOSComponent>unix</TargetOSComponent>
+    <TargetOSComponent Condition="'$(TargetOS)' == 'Windows_NT'">win</TargetOSComponent>
+    <TargetSpec>$(TargetOSComponent)-$(TargetArchitecture)</TargetSpec>
+
     <LibraryNamePrefix>lib</LibraryNamePrefix>
-    <LibraryNameExtension>.so</LibraryNameExtension>
-    <LibraryNameExtension Condition="'$(TargetsOSX)' == 'true'">.dylib</LibraryNameExtension>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetsWindows)' == 'true'">
-    <LibraryNamePrefix></LibraryNamePrefix>
-    <LibraryNameExtension>.dll</LibraryNameExtension>
+    <LibraryNamePrefix Condition="$([MSBuild]::IsOsPlatform('WINDOWS'))"></LibraryNamePrefix>
+    <LibraryNameExtension Condition="$([MSBuild]::IsOsPlatform('WINDOWS'))">.dll</LibraryNameExtension>
+    <LibraryNameExtension Condition="$([MSBuild]::IsOsPlatform('LINUX'))">.so</LibraryNameExtension>
+    <LibraryNameExtension Condition="$([MSBuild]::IsOsPlatform('OSX'))">.dylib</LibraryNameExtension>
+
+    <JitInterfaceLibraryName>$(LibraryNamePrefix)jitinterface$(LibraryNameExtension)</JitInterfaceLibraryName>
   </PropertyGroup>
 
   <ItemGroup>
-    <Content Include="$(BinDir)\$(LibraryNamePrefix)jitinterface$(LibraryNameExtension)"
+    <Content Include="$(BinDir)\$(JitInterfaceLibraryName)"
       CopyToOutputDirectory="PreserveNewest"
       CopyToPublishDirectory="PreserveNewest"
       Link="%(FileName)%(Extension)"
      />
 
-    <Content Include="$(BinDir)\$(LibraryNamePrefix)clrjit$(LibraryNameExtension)"
+    <Content Include="$(BinDir)\$(NativeArchFolder)$(LibraryNamePrefix)clrjit$(LibraryNameExtension)"
       CopyToOutputDirectory="PreserveNewest"
       CopyToPublishDirectory="PreserveNewest"
-      Link="$(LibraryNamePrefix)clrjitilc$(LibraryNameExtension)"
+      Link="$(LibraryNamePrefix)clrjit-$(TargetSpec)$(LibraryNameExtension)"
      />
   </ItemGroup>
 
+  <Target Name="CreateCrossTargetingPackage" AfterTargets="Build" Condition="'$(CrossHostArch)' != ''">
+
+    <PropertyGroup>
+      <CrossPackageFolder>$(BinDir)\$(CrossHostArch)\crossgen2</CrossPackageFolder>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackageFile Include="$(BinDir)\crossgen2\*" Exclude="$(BinDir)\crossgen2\$(JitInterfaceLibraryName)" />
```

Comment:
Should we also exclude `clrjit*`, which is an ARM64 binary?

Reply:
Fixed in 25th commit, thanks for pointing that out!